src/workerd/io/worker-entrypoint.c++

```diff
@@ -332,6 +341,14 @@ kj::Promise<void> WorkerEntrypoint::request(kj::HttpMethod method,
       failOpenService = context.getSubrequestChannelNoChecks(
           IoContext::NEXT_CLIENT_CHANNEL, false, kj::mv(cfBlobJson));
     }
+
+    if (proxyTask == kj::none && !loggedExceptionEarlier) {
+      KJ_IF_SOME(ctrl, abortController) {
+        context.addWaitUntil(context.run(
```

Comment:
TODO: I should avoid adding this task unless we know there is an event handler registered to receive the abort. I'm not quite sure where to do this though.

Reply:
A mechanism for this does not currently exist but we can look at adding it.

Reply:
Tricky part here is that you are trying to avoid taking the isolate lock, but the data structures you need to examine are protected by the lock. So we need to add some other mechanism where when an event handler is first added, we set some atomic boolean true, and you can check that atomic boolean here without the lock.

Reply:
Hmm, given that we only run this code when the request is actually canceled (not in the common case of completing successfully), maybe it's OK to just always take the lock after all. I was worried earlier that we'd be adding an extra lock on _every_ request, but that's not actually the case.

Reply:
I turned this into just a TODO comment for now